src/StorageReader/Exceptions.ts

```diff
@@ -12,33 +12,41 @@ export class NoMoreEntriesException extends Error {
 export class InvalidClaim extends Error {
   readonly ipfsFileHash: string
   readonly failureReason: FailureReason
+  readonly failureTime?: number
 
-  constructor(ipfsFileHash: string, failureReason: FailureReason) {
+  constructor(ipfsFileHash: string, failureReason: FailureReason, failureTime?: number) {
```

Comment:
Make `failureTime` default to `new Date().getTime()` so that the user of the error does not need to pass it in. Same goes for the other errors in this file and update the users so they do not pass it in.

Reply:
@kennylavender  Ok done!